File: stack/stack.ts
// Stack

class Stack {
  private stack: Array<number>;
  private stackSize: number;
  private lastIndex: number = -1;

  constructor (size: number) {
    this.stackSize = size;
    this.stack = new Array(size);
  }

  public push(ele: number): void {
    if (this.lastIndex + 1 >= this.stackSize) {
      return;
    }

    this.stack.push(ele);
    this.lastIndex = this.lastIndex + 1;
  };

  public pop(): number {
    if (this.lastIndex < 0) {
      return -1;
    }

    this.lastIndex = this.lastIndex - 1;
    return this.stack.pop();
  };
};

// main
(() => {
  const myStack = new Stack(10);

  myStack.push(1);
  myStack.push(2);

  console.log(myStack.pop());
})();

File: stack/Stack.java
package stack;

public class Stack {
    private int stack[];
    private int stackSize;
    private int lastIndex = -1;

    public Stack(int size) {
        this.stack = new int[size];
        stackSize = size;
    }

    public void push(int num) {
        if (this.lastIndex + 1 >= this.stackSize) {
            return;
        }

        this.lastIndex++;
        this.stack[this.lastIndex] = num;
    }

    public int pop() {
        if (this.lastIndex < 0) {
            return -1;
        }

        int num = this.stack[this.lastIndex];
        this.stack[this.lastIndex] = -1;
        this.lastIndex--;

        return num;
    }

    public static void main(String[] args) {
        Stack stack = new Stack(10);
        stack.push(1);
        stack.push(2);
        stack.push(3);
        stack.push(4);
        stack.push(5);
        System.out.println(stack.pop());
        System.out.println(stack.pop());
        System.out.println(stack.pop());
        System.out.println(stack.pop());
        System.out.println(stack.pop());
    }
}
